portal/trigger_states/models.py

```diff
@@ -0,0 +1,56 @@
+from datetime import datetime
+from sqlalchemy.dialects.postgresql import ENUM, JSONB
+from sqlalchemy.orm import make_transient
+
+from ..database import db
+from ..date_tools import FHIR_datetime
+
+
+trigger_state_enum = ENUM(
+    'unstarted', 'due', 'inprocess', 'processed', name='trigger_state_type',
+    create_type=False)
+
+
+class TriggerState(db.Model):
+    """ORM class for trigger state
+
+    Model patient's trigger state, retaining historical record for reporting.
+    NB only rows with `state` = `processed` include triggers.
+
+    """
+    __tablename__ = 'trigger_states'
+    id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.ForeignKey('users.id'), nullable=False, index=True)
+    state = db.Column('state', trigger_state_enum, nullable=False, index=True)
+    timestamp = db.Column(
+        db.DateTime, default=datetime.utcnow, nullable=False, index=True)
+    triggers = db.Column(JSONB)
```

Comment:
I think this is a list of FHIR references to QuestionnaireResponse resources- Why not a join table to the associated QuestionnaireResponses (or Observations)? Seems like this will be a commonly looked up value

Reply:
I've been waiting on the clarity, but agree that association with a QNR for a row with triggers makes the most sense.  A bit puzzled how to best handle the non-real-world situation where a user has multiple completed QNRs for a single "visit".